src/snowflake/snowpark/modin/plugin/extensions/utils.py

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+"""
+File containing utilities for the extensions API.
+"""
+from snowflake.snowpark.modin.utils import Fn
+
+cache_result_docstring = """
+Persists the current Snowpark pandas {object_name} to a temporary table to improve the latency of subsequent operations.
+
+Args:
+    inplace: bool, default False
+        Whether to perform the materialization inplace.
+
+Returns:
+    Snowpark pandas {object_name}
+
+Note:
+    - The temporary table produced by this method lasts for the duration of the session.
+
+Examples:
+{examples}
+"""
+
+cache_result_examples = """
+Let's make a {object_name} using a computationally expensive operation, e.g.:
+>>> {object_var_name} = {object_creation_call}
+
+Due to Snowpark pandas lazy evaluation paradigm, every time this {object_name} is used, it will be recomputed -
+causing every subsequent operation on this {object_name} to re-perform the 30 unions required to produce it.
+This makes subsequent operations more expensive. The `cache_result` API can be used to persist the
+{object_name} to a temporary table for the duration of the session - replacing the nested 30 unions with a single
+read from a table.
+
+>>> new_{object_var_name} = {object_var_name}.cache_result()
+
+>>> import numpy as np
+
+>>> np.all((new_{object_var_name} == {object_var_name}).values)
+True
+
+>>> from time import perf_counter
+
+>>> def time_reset_index({object_var_name}_obj):
+...     start = perf_counter()
+...     {object_var_name}_obj.reset_index(drop=True)
+...     end = perf_counter()
+...     return end - start
+...
+
+>>> uncached_time = time_reset_index({object_var_name})
+
+>>> cached_time = time_reset_index(new_{object_var_name})
+
+>>> assert cached_time < uncached_time
```

Comment:
```suggestion
```

I'd recommend not counting on a perform benchmark to pass every time we run this doctest.